hq/app/logic/SnykDisplay.scala

```diff
@@ -48,28 +52,49 @@ object SnykDisplay {
   def parseJsonToProjectIdList(s: String): Attempt[List[SnykProject]] =
     parseJsonToObject("project ids", s, body => (Json.parse(body) \ "projects").validate[List[SnykProject]])
 
-  def parseProjectVulnerabilitiesBody(s: String): Attempt[SnykProjectIssues] =
-    parseJsonToObject("project vulnerabilities", s, body => Json.parse(body).validate[SnykProjectIssues])
+  def parseOrganisationVulnerabilities(organisationAndResponseList: List[(SnykOrganisation, String)])(implicit ec: ExecutionContext): Attempt[List[SnykOrganisationIssues]] =
+    Attempt.traverse(organisationAndResponseList) { case (organisation, body) =>
+      parseOrganisationVulnerabilitiesBody(body).map(projectIssues => SnykOrganisationIssues(organisation, projectIssues))
+    }
+
+  def parseOrganisationVulnerabilitiesBody(s: String): Attempt[List[SnykProjectIssues]] =
+    parseJsonToObject("project vulnerabilities", s, body => {
+      val orgIssuesResult = (Json.parse(body) \ "results").validate[List[SnykProjectIssue]]
+      orgIssuesResult.map(groupProjectIssues)
+    })
 
-  def parseProjectVulnerabilities(bodies: List[String])(implicit ec: ExecutionContext): Attempt[List[SnykProjectIssues]] =
-    Attempt.traverse(bodies)(parseProjectVulnerabilitiesBody)
+  def groupProjectIssues(projectsAndIssues: List[SnykProjectIssue]): List[SnykProjectIssues] =
+    projectsAndIssues.groupBy(_.project).map { case (project, issues) =>
+      SnykProjectIssues(project, issues)
+    }.toList
 
-  def linkToSnykProject(snykProjectIssues: SnykProjectIssues, queryString: Option[String]): String = snykProjectIssues.project match {
-    case Some(project) if project.organisation.nonEmpty =>
-      s"https://snyk.io/org/${project.organisation.get.name}/project/${project.id}/${queryString.getOrElse("")}"
+  def linkToSnykProject(snykProjectIssues: SnykProjectIssues, org: SnykOrganisation, queryString: Option[String]): String = snykProjectIssues.project match {
+    case Some(project) =>
+      s"https://snyk.io/org/${org.name}/project/${project.id}/${queryString.getOrElse("")}"
```

Comment:
Something strange is happening with these links. It works for most cases, but not for ed tools and frontend?

Reply:
Sorry, missed this. I spotted this as well and put a bullet in the description. We need to use `slug` instead of `name` because you can rename orgs but the URLs are static.

I'm proposing we do this in a follow up because this PR has gotten pretty big already.

Reply:
ohh sorry, you did mention it in the description. Yep, follow up is fine!